woocommerce/woocommerce-ios
# Update “manage button” logic to cover all plans

Closes: https://github.com/woocommerce/woocommerce-ios/issues/10194
Closes: https://github.com/woocommerce/woocommerce-ios/issues/9916 

## Description
This PR addresses two issues regarding the "Manage your subscription" button:
1. The button would only load for sites with a current Essential Monthly plan, since we only checked for that specific case during M1.
2. The button would load for any site that is in the same account of a merchant with an active plan, since the check was performed account-wide, without taking into account the siteID (https://github.com/woocommerce/woocommerce-ios/issues/10194)

## Changes

We add a new `siteHasCurrentInAppPurchases(siteID: Int64)` action to the `InAppPurchaseStore` which will check if there's  any `VerificationResult<Transaction>` on the user account, which represents a current In-App Purchase. 

If there is, we decode the transaction token and extract the associated SiteID with that transaction. If the current store's siteID fits the extracted siteID, then we know the current site has a subscription via IAP.

## Testing instructions
For the following Woo Express stores. go to `Menu` > `Subscriptions` and observe that:
- On a Free Trial store with no upgrades, or expired, no button is rendered.
- On a store upgraded via web, no button is rendered.
- On a store upgraded via In-App Purchases, a "Manage your subscription" button is rendered. Tapping this button will behave as before, redirecting the merchant to the App Store `https://apps.apple.com/account/subscriptions` URL
